rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/are_all_terminal_heating_loops_purchased_heating.py

```diff
@@ -0,0 +1,63 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_exactly_one_with_field_value
+
+EXTERNAL_FLUID_SOURCE = schema_enums["ExternalFluidSourceOptions"]
+FLUID_LOOP_TYPE = schema_enums["FluidLoopOptions"]
+
+
+def are_all_terminal_heating_loops_purchased_heating(rmi_b, terminal_unit_id_list):
+    """Returns TRUE if the fluid loop associated with the heating_from_loop associated with each terminal unit is purchased heating. Returns FALSE if this is not the case.
+    ----------
+    rmi_b : json
+        RMD at RuleSetModelInstance level
+    terminal_unit_id_list : list
+        List of terminal units IDs
+    Returns
+    -------
+    bool
+        True: fluid loop associated with the heating_from_loop associated with each terminal unit is purchased heating
+        False: otherwise
+    """
+    are_all_terminal_heating_loops_purchased_heating_flag = True
+    purchased_heating_loop_list_b = []
+
+    external_fluid_sources = rmi_b.get("external_fluid_source")
+    if external_fluid_sources:
```

Comment:
An alternate way of getting this list would be:
```
purchased_heating_loop_list_b = [
    *find_all(f'external_fluid_source[*][?(@.type={EXTERNAL_FLUID_SOURCE.HOT_WATER})].loop, rmi_b),
    *find_all(f'external_fluid_source[*][?(@.type={EXTERNAL_FLUID_SOURCE.STEAM})].loop, rmi_b),
]
```

I am exploring an alternate Python jsonpath implementation that would further simplify something like this by allowing us to combine the two find_all expressions into one.
This alternate implementation is actively supported by Mark Borkum of PNNL.

Reply:
Unfortunately it seems jsonpath_ng does not support OR operator.
I did a slight change to make it work in one line, let me know if you found other package that works withe alternate jsonpath.

```python
    purchased_heating_loop_id_list_b = [*find_all(f'(external_fluid_source[?(@.type=={EXTERNAL_FLUID_SOURCE.HOT_WATER})].loop) | (external_fluid_source[?(@.type=={EXTERNAL_FLUID_SOURCE.STEAM})].loop)', rmi_b)]
```

Reply:
Since you've combined the two find_all's into one, you no longer need to wrap it with [* ... ].

Correct, jsonpath-ng does not support an OR operator.
https://github.com/pacifica/python-jsonpath2 does support OR (most conveniently as a comma).  With this package, the find_all would be:
```
find_all(f"external_fluid_source[?(@.type=={EXTERNAL_FLUID_SOURCE.HOT_WATER}, @.type=={EXTERNAL_FLUID_SOURCE. STEAM})].loop")
```

Reply:
Does the jsonpath2 supports AND operator? I find some use cases that may need to intersect two jsonpaths.

Reply:
Yes. Although I think we can do that using jsonpath-ng too by appending restrictions `[?(one_restriction)][?(another_restriction)]`